specs/core/1_custody-game.md

```diff
@@ -464,7 +482,18 @@ def process_bit_challenge_response(state: BeaconState,
 
 ## Per-epoch processing
 
-Run `process_challenge_deadlines(state)` immediately after `process_ejections(state)`:
+### Handling of custody-related deadlines
+
+Run `process_reveal_deadlines(state)` immediately after `process_ejections(state)`:
+
+```python
+def process_reveal_deadlines(state) -> None:
+    for idx, validator in enumerate(state.validator_registry):
+        if validator.latest_custody_reveal_period + 1 < get_validators_current_custody_reveal_period(state, idx):
+            slash_validator(state, idx)
```

Comment:
This looks dangerous. This basically means that being offline for ~9 days will get you slashed, whereas the standard in the rest of the protocol is that you need for be offline for ~73 days for that to happen. So I think it needs to be `+ CUSTODY_RESPONSE_DEADLINE // EPOCHS_PER_CUSTODY_PERIOD`, not `+ 1`.

Reply:
Yes, that sounds reasonable. Just a question, do we want to set any deadlines for bitfield challenges? I just checked the logic, and currently there are none. We probably don't want a logic to remember when reveals happened, so it would make sense to just set this to 2 * CUSTODY_RESPONSE_DEADLINE?

Reply:
> This looks dangerous. This basically means that being offline for ~9 days will get you slashed, whereas the standard in the rest of the protocol is that you need for be offline for ~73 days for that to happen. So I think it needs to be `+ CUSTODY_RESPONSE_DEADLINE // EPOCHS_PER_CUSTODY_PERIOD`, not `+ 1`.

Agreed and implemented.

There are deadlines for bitfield challenges. See line 506 below

Reply:
> There are deadlines for bitfield challenges. See line 506 below

This deadline concerns the response to the bit challenge. I am suggesting we should not allow bit challenges arbitrarily into the past (see my comment on line 380)